scripts/fill_dcp1_file_metadata/fill_dcp1_metadata.py

```diff
@@ -0,0 +1,188 @@
+import requests as rq
+import argparse
+import json
+import logging
+
+from requests.exceptions import HTTPError
+
+from hca_ingest.api.ingestapi import IngestApi
+
+"""
+The script follows the next steps to fix the missing size and fileContentType for DCP1 datasets:
+
+1. Check the project UUID provided is valid 
+2. Retrieve all data file metadata from azul
+3. Parse and massage the data to avoid DCP/1 / HCA Release / DCP/2 analysis files (Not in ingest)
+4. For each of the metadata files:
+   a. Check metadata is not already filled (Priority: Ingest > Azul)
+   b. PUT metadata file
+"""
+INGEST_TOKEN = ""
+
+# TODO: Check if file metadata is editable (submission editable). Don't know how to do it without 1 API call/file
+
+
+class NotPublishedUuid(HTTPError):
+    def __init__(self, response: rq.Response, uuid: str):
+        super().__init__(f"UUID {uuid} not found in Azul: GET request errored with status code {response.status_code}")
+
+
+def is_published_uuid(uuid: str) -> str:
+    """
+    Check if the UUID provided corresponds to a published Azul project
+    :param uuid:    UUID of the project which contains the files
+    :return uuid:
+    """
+    r = rq.get(f"https://service.azul.data.humancellatlas.org/index/projects/{uuid}")
+    try:
+        r.raise_for_status()
+    except HTTPError:
+        raise NotPublishedUuid(r, uuid)
+    return uuid
+
+
+def define_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-p", "--project-uuid", type=is_published_uuid, required=True,
+                        help="UUID of the project to be patched. Needs to be published in the DCP.")
+    parser.add_argument("-d", "--dry-run", action="store_true", default=False, help="Dry-run flag. No PATCH operations"
+                                                                                    "will be performed.")
+
+    return parser
+
+
+def build_azul_query(project_uuid: str):
+    """
+    Build a ElasticSearch query to extract all metadata related to project with UUID "project_uuid"
+    :param project_uuid:
+    :return query:          string dump of query
+    """
+    query = {
+        "projectId": {
+            "is":
+                [
+                    project_uuid
+                ]
+        }
+    }
+    return json.dumps(query)
+
+
+def get_files_metadata_from_azul(query: str) -> list:
+    """
+    Retrieve file metadata from azul
+    :param query:           Query in str format to pass as a filter
+    :return files:          List of file metadata
+    """
+    # Build URL/endpoint and request metadata
+    azul_base = "https://service.azul.data.humancellatlas.org/"
```

Comment:
I would have the azul url as a config element. This is the proper "future ready" style, making you ready for hostname change, without having to modify a lot of code.
See [an example in geo_to_hca](https://github.com/ebi-ait/geo_to_hca/blob/master/geo_to_hca/config.py)